models/src/anemoi/models/preprocessing/postprocessor.py

```diff
@@ -0,0 +1,354 @@
+
+    def _validate_indices(self):
+        assert (
+            len(self.index_training_output) == len(self.index_inference_output) <= len(self.postprocessorfunctions)
+        ), (
+            f"Error creating imputation indices {len(self.index_training_output)}, "
+            f"{len(self.index_inference_output)}, {len(self.postprocessorfunctions)}"
+        )
+
+    def _create_imputation_indices(
+        self,
+    ):
+        """Create the indices for imputation."""
+        name_to_index_training_output = self.data_indices.data.output.name_to_index
+        name_to_index_inference_output = self.data_indices.model.output.name_to_index
+
+        self.num_training_output_vars = len(name_to_index_training_output)
+        self.num_inference_output_vars = len(name_to_index_inference_output)
+
+        (
+            self.index_training_output,
+            self.index_inference_output,
+            self.postprocessorfunctions,
+        ) = ([], [], [])
+
+        # Create indices for imputation
+        for name in name_to_index_training_output:
+
+            method = self.methods.get(name, self.default)
+            if method == "none":
+                LOGGER.debug(f"Postprocessor: skipping {name} as no postprocessing method is specified")
+                continue
+
+            assert name in name_to_index_inference_output, (
+                f"Postprocessor: {name} not found in inference output indices. "
+                f"Postprocessors cannot be applied to foorcing variables."
+            )
+
+            self.index_training_output.append(name_to_index_training_output.get(name, None))
+            self.index_inference_output.append(name_to_index_inference_output.get(name, None))
+
+            if method == "relu":
+                self.postprocessorfunctions.append(torch.nn.functional.relu)
+            elif method == "hardtanh":
+                self.postprocessorfunctions.append(torch.nn.functional.hardtanh)
+
+            LOGGER.info(f"Postprocessor: applying {method} to {name}")
+
+    def inverse_transform(self, x: torch.Tensor, in_place: bool = True) -> torch.Tensor:
+        """Impute missing values in the input tensor."""
+        if not in_place:
+            x = x.clone()
+
+        if x.shape[-1] == self.num_training_output_vars:
+            index = self.index_training_output
+        elif x.shape[-1] == self.num_inference_output_vars:
+            index = self.index_inference_output
+        else:
+            raise ValueError(
+                f"Input tensor ({x.shape[-1]}) does not match the training "
+                f"({self.num_training_output_vars}) or inference shape ({self.num_inference_output_vars})",
+            )
+
+        # Replace values
+        for postprocessor, idx_dst in zip(self.postprocessorfunctions, index):
+            if idx_dst is not None:
+                x[..., idx_dst] = postprocessor(x[..., idx_dst])
+        return x
+
+
+class CustomRelu(torch.nn.Module):
```

Comment:
activations functions should live under `/models/layers/activations.py` for consistency